v2.0/performance-tuning.md

```diff
@@ -0,0 +1,1275 @@
+|  │    └── render                   |           |                                                                          |
+|  │         └── scan                |           |                                                                          |
+|  │                                 | table     | rides@primary                                                            |
+|  │                                 | spans     | ALL                                                                      |
+|  └── subquery                      |           |                                                                          |
+|       │                            | id        | @S1                                                                      |
+|       │                            | sql       | (SELECT vehicle_id FROM rides GROUP BY vehicle_id ORDER BY count(*) DESC |
+|                                    |           | LIMIT 5)                                                                 |
+|       │                            | exec mode | all rows normalized                                                      |
+|       └── limit                    |           |                                                                          |
+|            └── sort                |           |                                                                          |
+|                 │                  | order     | -count                                                                   |
+|                 │                  | strategy  | top 5                                                                    |
+|                 └── group          |           |                                                                          |
+|                      │             | group by  | @1-@1                                                                    |
+|                      └── render    |           |                                                                          |
+|                           └── scan |           |                                                                          |
+|                                    | table     | rides@primary                                                            |
+|                                    | spans     | ALL                                                                      |
++------------------------------------+-----------+--------------------------------------------------------------------------+
+(21 rows)
+~~~
+
+This is a complex query plan, but the important thing to note is the full table scan of `rides@primary` above the `subquery`. This shows you that, after the subquery returns the IDs of the top 5 vehicles, CockroachDB scans the entire primary index to find the rows with `max(end_time)` for each `vehicle_id`, although you might expect CockroachDB to more efficiently use the secondary index on `vehicle_id`.
```

Comment:
Mention that we're working to lift this limitation in a future version.

Reply:
Done.